packages/space-opera/src/components/model_viewer_snippet/model_viewer_snippet.ts

```diff
@@ -0,0 +1,178 @@
+
+/**
+ * Export panel.
+ */
+@customElement('me-export-panel')
+export class ExportPanel extends ConnectedLitElement {
+  @internalProperty() config: ModelViewerConfig = {};
+  @internalProperty() hotspots: HotspotConfig[] = [];
+  @internalProperty() camera: Camera = INITIAL_CAMERA;
+
+  @query('snippet-viewer') snippetViewer!: SnippetViewer;
+  @query('me-export-zip-button') exportZipButton!: ExportZipButton;
+
+  stateChanged(state: State) {
+    this.config = state.config;
+    this.camera = state.camera;
+    this.hotspots = state.hotspots;
+  }
+
+  render() {
+    const editedConfig = {...this.config};
+    applyCameraEdits(editedConfig, this.camera);
+    // Don't render the data URI. Users should download the edited GLB and put
+    // the right URL here.
+    editedConfig.src = `model.glb`;
+    const snippet =
+        renderModelViewer(editedConfig, {}, renderHotspots(this.hotspots));
+
+    return html`
+    <me-expandable-tab tabName="Export" .open=${true}>
+    <div slot="content">
+      <me-download-button id="download-gltf"></me-download-button>
+      <me-export-zip-button id="export-zip"></me-export-zip-button><br/>
+      <br/>
+      And use this model-viewer snippet:<br/>
+      <br/>
+      <snippet-viewer .renderedSnippet=${snippet}
+        .renderedStyle=${this.hotspots.length > 0 ? hotspotStyles.cssText : ``}>
+      </snippet-viewer>
+    </div>
+    </me-expandable-tab>`;
+  }
+
+  protected updated() {
+    this.snippetViewer.updateComplete.then(() => {
+      this.exportZipButton.snippetText =
+          this.snippetViewer.snippet.textContent || '';
+    });
+  }
+}
+
+/**
+ * Import/Export panel.
+ * TODO:: This should be factored out/renamed.
+ */
+@customElement('model-viewer-snippet')
+export class ModelViewerSnippet extends LitElement {
+  static get styles() {
+    return css`
+  #mv-input {
+    width: 95%;
+  }
+        `;
+  }
+
+  @query('textarea#mv-input') private readonly textArea!: HTMLInputElement;
+
+  async handleSubmitSnippet(event: Event) {
+    event.preventDefault();
+    if (!this.textArea) return;
+    const textArea = this.textArea;
+    const inputText: string = textArea.value.trim();
+    if (inputText.match(
+            /<\s*model-viewer[^>]*\s*>(\n|.)*<\s*\/\s*model-viewer>/)) {
+      // Change background color on valid/invalid input.
+      textArea.style.backgroundColor = 'white';
+      const config = parseSnippet(inputText);
+
+      const hotspotErrors: Error[] = [];
+      const hotspotConfigs = parseHotspotsFromSnippet(inputText, hotspotErrors);
+      // TODO:: Properly display errors to user
+      for (const error of hotspotErrors) {
+        console.warn(error);
+      }
+
+      try {
+        // TODO: Because of update-batching, we need to sleep in
+        // order to force a GLB reload. Better way?
+        dispatchGltfUrl(undefined);
+        await new Promise(resolve => {
+          setTimeout(resolve, 100);
```

Comment:
0 does work. And, the proper solution would be to add a timestamp-ish value to state, ie. "please reload the GLB even tho the URL is the same..because the user clicked it AGAIN"

Reply:
Updated this to use 0